### PR TITLE
Do not allow empty Rego source strings on ConstraintTemplates

### PR DESCRIPTION
**Pull request by maxsmythe**
It causes a panic

Signed-off-by: Max Smythe <smythe@google.com>

### CODE REVIEW

**Comment by maxsmythe** [comment]
This will partially address https://github.com/open-policy-agent/gatekeeper/issues/238 when merged into GK.

I'd still like the version info so that I can use line numbers to fix the nil pointer dereference closer to the source (if possible).

**Comment by dolv** [comment]
quay.io/open-policy-agent/gatekeeper:v3.0.4-beta.0

**Comment by maxsmythe** [comment]
Thanks! Found the root of the nil-dereference and removed it :)

File: constraint/pkg/client/rego_helpers.go [hunk 1]
```diff
@@ -143,10 +149,16 @@ type ruleArities map[string]int
...
 func requireRules(name, rego string, reqs ruleArities) error {
+	if rego == "" {
+		return errors.New("Rego source code is empty")
+	}
 	module, err := ast.ParseModule(name, rego)
 	if err != nil {
 		return err
 	}
+	if module == nil {
+		return "", errors.New("Module could not be parsed")
```
Comment (ritazh) [inline constraint/pkg/client/rego_helpers.go]: oh this is giving us an error. should just return error, no string

Comment (maxsmythe) [reply constraint/pkg/client/rego_helpers.go]: oops, that's what I get for not using an IDE...

fixed, will push if tests pass.

**Pull request merged**